tests/e2e_test/module_deletion_test.go

```diff
@@ -0,0 +1,187 @@
+		v1beta2.SyncStrategyLocalSecret)
+	GinkgoWriter.Printf("kyma before create %v\n", kyma)
+
+	BeforeAll(func() {
+		// make sure we can list Kymas to ensure CRDs have been installed
+		err := controlPlaneClient.List(ctx, &v1beta2.KymaList{})
+		Expect(meta.IsNoMatchError(err)).To(BeFalse())
+	})
+
+	Context("Given a Kyma Cluster", func() {
+		It("When its corresponding CR is created", func() {
+			Eventually(CreateKymaSecret).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient).
+				Should(Succeed())
+			Eventually(controlPlaneClient.Create).
+				WithContext(ctx).
+				WithArguments(kyma).
+				Should(Succeed())
+			By("Then the Kyma CR is in a `Ready` state on control-plane cluster")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+				Should(Succeed())
+			By("And the Kyma CR is in `Ready` sate on runtime cluster")
+			Eventually(CheckRemoteKymaCR).
+				WithContext(ctx).
+				WithArguments(remoteNamespace, []v1beta2.Module{}, runtimeClient, v1beta2.StateReady).
+				Should(Succeed())
+		})
+		It("When a module named `template-operator` is enabled", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", "regular", runtimeClient).
+				Should(Succeed())
+			By("Then the template-operator manager is deployed on runtime cluster")
+			Eventually(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system", "apps/v1",
+					"Deployment", runtimeClient).
+				Should(Succeed())
+			By("And the Kyma CR is in a `Ready` state")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+				Should(Succeed())
+		})
+
+		It("When the module `template-operator` is disabled with an existing finalizer", func() {
+			Eventually(SetFinalizer).
+				WithContext(ctx).
+				WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+					[]string{"sample.kyma-project.io/finalizer", "blocking-finalizer"}, runtimeClient).
+				Should(Succeed())
+			Eventually(DisableModule).
+				WithContext(ctx).
+				WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", runtimeClient).
+				Should(Succeed())
+			By("Then Kyma CR is in state `Processing`")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateProcessing).
+				Should(Succeed())
+			By("And Manifest CR is in state `Deleting`")
+			Eventually(CheckManifestIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), "template-operator", controlPlaneClient, v2.StateDeleting).
+				Should(Succeed())
+			By("And Sample CR is not removed")
+			Eventually(CheckIfExists).
```

Comment:
```suggestion
			Consistently(CheckIfExists).
```

Reply:
https://github.com/kyma-project/lifecycle-manager/pull/912#discussion_r1350506044
the second point I mentioned there is not covered, SampleCR should verify in `Deleting` state and deletiontimestamp not zero